eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -15,24 +15,47 @@ parameters:
 #   "ProvisionerApplicationId": "<provisioner app id>",
 #   "ProvisionerApplicationSecret": "<provisioner app secret>",
 #   "Environment": "AzureCloud | AzureGov | AzureChina | <other environment>"
+#   "EnvironmentVariables": {
+#       "SERVICE_MANAGEMENT_URL": "<service management url>",
+#       "STORAGE_ENDPOINT_SUFFIX": "<storage endpoint suffix>",
+#       "RESOURCE_MANAGER_URL": "<resource manager url>",
+#       "SEARCH_ENDPOINT_SUFFIX": "<search endpoint suffix>",
+#       "COSMOS_TABLES_ENDPOINT_SUFFIX": "<cosmos tables endpoint suffix>"
+#   },
+#   "ArmTemplateParameters": {
+#       "keyVaultDomainSuffix": "<keyVaultDomainSuffix>",
+#       "storageEndpointSuffix": "<storageEndpointSuffix>",
+#       "endpointSuffix": "<endpointSuffix>",
+#       "azureAuthorityHost": "<azureAuthorityHost>",
+#       "keyVaultEndpointSuffix": "<keyVaultEndpointSuffix>"
+#   }
 # }
 
+
 steps:
   - template: /eng/common/TestResources/setup-az-modules.yml
 
   - pwsh: |
       eng/common/TestResources/Import-AzModules.ps1
 
-      $subscriptionConfiguration = @"
+      $subscriptionConfiguration = @'
         ${{ parameters.SubscriptionConfiguration }}
-      "@ | ConvertFrom-Json -AsHashtable;
+      '@ | ConvertFrom-Json -AsHashtable;
+
+      if (!$subscriptionConfiguration.ArmTemplateParameters) {
+        $subscriptionConfiguration.ArmTemplateParameters = @{}
+      }
+
+      $armTemplateParameters = ${{ parameters.ArmTemplateParameters }}
```

Comment:
Doesn't this need to be declared a parameter? And in the parent template as well so I can pass it through from sdk/keyvault/tests.yml? Or is declaring parameters not *required*? Even still, it makes them more discoverable.

Reply:
ArmTemplateParameters is declared as a parameter at the top of this file. Or are you referring to something else?

It will get passed through like so, with ArmTemplateParameters coming from a variable declared in AdditionalPlatforms.

```
      - template: /eng/common/TestResources/deploy-test-resources.yml
        parameters:
          Location: ${{ parameters.CloudConfig.Location }}
          ServiceDirectory: '${{ parameters.ServiceDirectory }}'
          SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
          ArmTemplateParameters: $(ArmTemplateParameters)
```

So you would need to declare ArmTemplateParameters on your platform definition in tests.yml, and we pass it through from the matrix entry. I haven't pushed the PR in azure-sdk-for-net yet since I'm still working out an issue with when the var is empty.